File: data_structures/stacks/stack_upon_array/stack.js
class Stack {
    constructor() {
        this.data = [];
    }

    pop() {
        return this.isEmpty()? null: this.data.pop();
    }

    push(value) {
        this.data.push(value);
        return this;
    }

    peek() {
        return this.isEmpty()? null: this.data[this.data.length - 1];
    }

    isEmpty() {
        return this.data.length === 0;
    }

    printAsList() {
        console.log(this.data);
    }
}

File: data_structures/stacks/stack_upon_array/stack.py
class Stack:
    def __init__(self):
        self.data = []

    def pop(self):
        if self.is_empty():
            return None
        return self.data.pop()

    def push(self, value):
        return self.data.append(value)

    def peek(self):
        return self.data[len(self.data) - 1] if self.data else None

    def is_empty(self):
        return len(self.data) == 0

    def print_as_list(self):
        print(self.data)